runtest.py

```diff
@@ -111,255 +122,186 @@ def kill_legacy_process(self):
+                    if test_case:
+                        # case setup
+                        self.case_setup()
+
+                        # case run
+                        runtime_case_data = {}
+                        test_case_module_name = ""
+                        test_case_fp = ""
+                        test_env = None
+                        test_name = ""
+                        if self.index_config['case-type'] == "pt":
+                            test_case_module_name = "%s.%s" % (self.global_config['default-test-dn'],  "test_pilot_run")
+                            runtime_case_data["SIKULI_SCRIPT_PATH"] = test_case
+                            runtime_case_data["TEST_SCRIPT_PY_DIR_PATH"] = os.sep.join(test_case_module_name.split(".")[:-1])
+                            if test_case.endswith(os.sep):
+                                test_name = test_case.split(os.sep)[-2].split(".")[0]
+                            else:
+                                test_name = test_case.split(os.sep)[-1].split(".")[0]
+                            test_case_fp = test_case
                         else:
-                            test_name = test_case.split(os.sep)[-1].split(".")[0]
-                    else:
-                        test_case_fp = test_case.replace(".", os.sep) + ".py"
-                        test_name = test_case.split(".")[-1]
-                        case_data[test_case]["TEST_SCRIPT_PY_DIR_PATH"] = os.sep.join(test_case.split(".")[:-1])
-                        if os.path.exists(test_case_fp):
+                            test_case_fp = test_case.replace(".", os.sep) + ".py"
+                            test_name = test_case.split(".")[-1]
                             test_case_module_name = test_case
-                            test_env = self.get_test_env(**case_data[test_case])
+                            runtime_case_data["TEST_SCRIPT_PY_DIR_PATH"] = os.sep.join(test_case.split(".")[:-1])
+                            # if webdriver is enable, we need to get parameters for running browsers
+                            if self.exec_config['user-simulation-tool'] == self.global_config[
+                                'default-user-simulation-tool-webdriver']:
+                                runtime_case_data['webdriver'] = "True"
+                                runtime_case_data['browser_type'] = self.exec_config['webdriver-run-browser']
+                        if os.path.exists(test_case_fp):
+                            test_env = self.get_test_env(**runtime_case_data)
                         else:
                             self.logger.error("Test script [%s] is not exist!" % test_case_fp)
-                            test_env = None
-
-                    # if webdriver is enable, we need to get parameters for running browsers
-                    if self.webdriver['enable']:
-                        test_env['webdriver'] = "True"
-                        if self.webdriver['run_firefox']:
```

Comment:
webdriver original default json
{
  "webdriver": {
    "enable": true,
    "run_firefox": true,
    "run_chrome": false
  }
}

Original Line 309-314 would run:
both browser: if both run_firefox and run_chrome set to true
firefox: if run_firefox set to true only
chrome: if run_chrome set to true only

It should be handled here or in loop_test(). Otherwise, the webdriver test case would be only be run in one browser.